dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java

```diff
@@ -479,81 +498,187 @@ public static void applyJsonFields(String value) {
+                        Tr.warning(tc, "JSON_FIELDS_NO_MATCH");
+                    }
+                    valueFound = false;//reset valueFound boolean
+
+                } else if (entry.length == 3) {
+                    entry[1] = entry[1].trim();
+                    entry[2] = entry[2].trim();
+                    //add properties to their respective hashmaps and trim whitespaces
+                    if (CollectorConstants.MESSAGES_CONFIG_VAL.equals(entry[0])) {
+                        if (LogTraceList.contains(entry[1]) || entry[1].startsWith("ext_")) {
+                            messageMap.put(entry[1], entry[2]);
+                            valueFound = true;
+                        }
+                    } else if (CollectorConstants.TRACE_CONFIG_VAL.equals(entry[0])) {
+                        if (LogTraceList.contains(entry[1]) || entry[1].startsWith("ext_")) {
+                            traceMap.put(entry[1], entry[2]);
+                            valueFound = true;
+                        }
+                    } else if (CollectorConstants.FFDC_CONFIG_VAL.equals(entry[0])) {
+                        if (FFDCList.contains(entry[1])) {
+                            ffdcMap.put(entry[1], entry[2]);
+                            valueFound = true;
+                        }
+                    } else if (CollectorConstants.ACCESS_CONFIG_VAL.equals(entry[0])) {
+                        if (AccessLogList.contains(entry[1])) {
+                            accessLogMap.put(entry[1], entry[2]);
+                            valueFound = true;
+                        }
+                    } else if (CollectorConstants.AUDIT_CONFIG_VAL.equals(entry[0])) {
+                        if (AuditList.contains(entry[1])) {
+                            auditMap.put(entry[1], entry[2]);
+                            valueFound = true;
+                        }
+                    } else {
+                        isInvalidField = true;
+                        Tr.warning(tc, "JSON_FIELDS_INCORRECT_EVENT_TYPE");
+                    }
+                    if (!valueFound && !isInvalidField) {
+                        //if the value does not exist in any of the known keys, give a warning
+                        Tr.warning(tc, "JSON_FIELDS_NO_MATCH");
+                    }
+                    valueFound = false;
+                    isInvalidField = false;
                 } else {
-                    Tr.warning(tc, "JSON_FIELDS_INCORRECT_EVENT_TYPE");
+                    Tr.warning(tc, "JSON_FIELDS_FORMAT_WARNING_2");
                 }
-                if (!valueFound) {
-                    //if the value does not exist in any of the known keys, give a warning
-                    Tr.warning(tc, "JSON_FIELDS_NO_MATCH");
-                }
-                valueFound = false;
-            } else {
-                Tr.warning(tc, "JSON_FIELDS_FORMAT_WARNING_2");
             }
         }
```

Comment:
Store omitFieldsMap as an ArrayList for each LogData type, since it's faster to look up as an ArrayList when omitting fields.